src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,138 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private static final double EXPAND_MULTIPLIER = 1.5;
+    private T[] collection;
 
-    @Override
-    public void add(T value) {
+    private int size = 0;
 
+    public ArrayList() {
+        this.collection = (T[]) new Object[CAPACITY];
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T value) {
+        if (size == collection.length) {
+            expand();
+        }
+        collection[size] = value;
+        size++;
+    }
 
+    @Override
+    public void add(T value, int index) throws ArrayIndexOutOfBoundsException {
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException("Index out of bound");
+        }
+        if (size == collection.length) {
+            expand();
+        }
```

Comment:
you are expanding here collection array, but later reassign it to the array which length = size + 1;